apps/timetable/ui/js/index.js

```diff
@@ -65,26 +55,19 @@ define(['gh.core', 'gh.subheader', 'gh.calendar', 'gh.student-listview', 'jquery
      * @private
      */
     var setUpCalendar = function() {
+        // Render the calendar template
         gh.api.utilAPI.renderTemplate($('#gh-calendar-template'), {
             'data': null
         }, $('#gh-main'));
 
-        if (gh.data.me.anon) {
-            $(document).trigger('gh.calendar.init');
-        } else {
-            var range = gh.api.utilAPI.getCalendarDateRange();
-            gh.api.userAPI.getUserCalendar(gh.data.me.id, range.start, range.end, function(err, data) {
-                if (err) {
-                    gh.api.utilAPI.notification('Fetching user calendar failed.', 'An error occurred while fetching the user calendar.', 'error');
-                }
+        // Initialise the calendar
+        $(document).trigger('gh.calendar.init', {'triposData': triposData});
 
-                var calendarData = {
-                    'triposData': triposData,
-                    'events': data
-                };
+        // Fetch the user's events
+        if (!gh.data.me.anon) {
 
-                $(document).trigger('gh.calendar.init', calendarData);
-            });
+            // Put the calendar on today's view
+            $(document).trigger('gh.calendar.navigateToToday');
```

Comment:
It's true that the calendar automatically displays the current date, but the events aren't fetched by default. I explicitly trigger this event here so that all the events are fetched for the user. This oneliner replaces the previous callback jibba jabba. And we can only do this when the user is logged in...